app/src/main/java/com/github/swent/echo/viewmodels/HomeScreenViewModel.kt

```diff
@@ -66,18 +66,24 @@ constructor(
             )
         )
     val filtersContainer = _filtersContainer.asStateFlow()
-
-    private val _profileName =
-        MutableStateFlow<String>("John Doe") // placeholder until we have user profiles
+    private val _profileName = MutableStateFlow<String>("")
     val profileName = _profileName.asStateFlow()
-    private val _profileClass =
-        MutableStateFlow<String>("IN - BA6") // placeholder until we have user profiles
+    private val _profileClass = MutableStateFlow<String>("")
     val profileClass = _profileClass.asStateFlow()
+    private var section = ""
+    private var semester = ""
 
     init {
         viewModelScope.launch {
+            val userId = authenticationService.getCurrentUserID() ?: ""
```

Comment:
As you are in the home screen, you can safely assume that the current user id exists, e.g. `val userId = authenticationService.getCurrentUserID()!!`.